Agenda for 2025-02-05 CG meeting (#710)
solid/specification: meetings/2025-02-05.md

* RG: We are not telling the WG to include this, but we are making them aware that there are some problems (?)
* Jesse: If you have a clear solution in mind, no one is going to object that, and the WG will take this as a valuable input.
* ...
* eP: If someone has a clear idea about why blank nodes should be allowed with N3 patch, they should create a PR so we can discuss on concrete terms with the community. And maybe warn the WG about this change.
* Jesse: My proposal is to delete the "MUST NOT"
* HZ: Minimal change could change *MUST NOT* to *SHOULD NOT*

ACTION: Jesse will create a PR removing that line, he will link to this discussion.

### Misc

* SR: I want to post discussion about profile management. It seems that Solid didn't think about it. The profiles are public, there is also a need for private profiles. If you share the full profile it can be problematic. The profile is the most sensitive data on the pod.
* JW: There is a notion of extended profiles that are used by implementaiton. In public profile there are minimal information, there is a pointer to extended profile which are access protected.

> * SR: It is not only about public/private but about fine grained access. For now it is a single resource.

Reviewer: Thanks for the links. Yes I will post a message on the Solid forum about this subject. I briefly mentionned this topic at the end of the meeting as an intro, but there is much more to say.

> * SR: It is not only about public/private but about fine grained access. For now it is a single resource.

Reviewer 2: Eventually significant considerations or changes need to make their way into the Solid CG's GitHub or mailing list (or CG meetings), so why not create an issue or a PR? They can be cross-linked and classified on GitHub alongside other related issues.

* If this is your first time, welcome! please introduce yourself.

### Scribes
* Sebastien Rosset
* Rui Zhao

### Introductions
* Areeba Aziz: From Canada, work at Amazon, really interested in the mission of Solid. Joined to get to know everyone.

---

## Topics

### Update on LWS Use Cases

> * HZ: We've collected 100+ use cases, and we are now moving to a second phase of infering requirement out of them. 
>   We will be updating the use cases with a glossary of terms.
>   We came up with a common context for all LWS issues .
>   We'll accept use cases for months to come, so feel free to add more as necessary.

Reviewer 3: ```suggestion
* HZ: We've collected 100+ use cases, and we are now moving to a second phase of inferring requirements out of them. 
  We will be updating the use cases with a glossary of terms.
  We came up with a common context for all LWS issues.
  We'll accept use cases for months to come, so feel free to add more as necessary.
```

> * eP: Are there are use cases that go beyond the scope of LWS, that would fit more into the CG, like interop ?
> * HZ: IMO absolutely. It is quite probable that the WG is not going to take all requirements.
>   The question translates to what will be in the spec, and the scope of the (WG) group. Maybe we will create a sub-group. And we may ask help to the CG. We will get into this in a couple of months.

Reviewer 3: ```suggestion
* eP: Are there are use cases that go beyond the scope of LWS, that would fit more into the CG, like interop?
* HZ: IMO absolutely. It is quite probable that the WG is not going to take all requirements.
  The question translates to what will be in the spec, and the scope of the LWS Working Group. Maybe we will create a sub-group. We may ask for help from the Solid CG. We will get into this in a couple of months.
```

> * eP: Will the WG and CG work on the common set of use cases? The requirements could fall either under WG or CG.
> * HZ: The work is public, with issues, etc. So feel free to engage a conversation there.
> * RZ: Some of my use cases don't fit into the current categories. Where should this be discussed ?
> * HZ: You can do that in the issues.
>   Personally I would like to consider Niko's proposal to decouple protocols (HTTP, etc) but I don't know yet if this is going to enter as a requirement for the spec. We're going to discuss this within the WG.

Reviewer 3: ```suggestion
* eP: Will the LWS WG and Solid CG work on a common set of use cases? The requirements could fall either under LWS WG or Solid CG.
* HZ: The work is public, with issues, etc. So feel free to engage a conversation there.
* RZ: Some of my use cases don't fit into the current categories. Where should this be discussed?
* HZ: You can do that in the issues.
  Personally I would like to consider Niko's proposal to decouple protocols (HTTP, etc.), but I don't know yet if this is going to enter as a requirement for the spec. We're going to discuss this within the LWS WG.
```

>   Meetings take place every Monday morning US time, have a look at the minutes https://www.w3.org/groups/wg/lws/calendar/?past=1&tf=0
> * Rahul: WG chairs can and is typically expected to invite "experts" for this kind of issues. You can also ask to be invited. 
> * Rahul: You can nominate yourself to be an "invited expert". This may concern Sébastien in particular.

Reviewer 3: ```suggestion
  Meetings take place every Monday morning, US time. Have a look at the minutes — <https://www.w3.org/groups/wg/lws/calendar/?past=1&tf=0>.
* Rahul: WG chairs can and are typically expected to invite "experts" for this kind of issues. You can also ask to be invited. 
* Rahul: You can nominate yourself to be an "invited expert". This may concern Sébastien in particular.
```

### [New Work Item] ActivityPub/Fediverse interoperability

https://github.com/solid/specification/issues/708

> * eP: I will have a call with Sebastien right after this call. We could use few minutes in that call to coordinate next steps.
> * eP: I would like to decide if we take it as a group item. If we create a repo on Friday social web CG will have a meeting. As PAC recommended last week, we should coordinate with the Social Web group.
> * SR: nothing to add, it would be great if a repository could be created.
> * HZ: we can make the decision now. Any objections? (None)
> * Rahul: It's a good idea to formalize that with a repo.
> 
> ACTION: elf Pavlik and Hadrian to work with W3C and Jesse to get the solid/activitypub-interop repository created.

Reviewer 3: ```suggestion
* eP: I will have a call with Sébastien right after this call. We could use few minutes in that call to coordinate next steps.
* eP: I would like to decide whether we take it as a group item. Whether we create a repo. On Friday, Social Web CG will have a meeting. As PAC recommended last week, we should coordinate with the Social Web group.
* SR: Nothing to add. It would be great if a repository could be created.
* HZ: We can make the decision now. Any objections?
  (None)
* Rahul: It's a good idea to formalize that with a repo.

ACTION: elf Pavlik and Hadrian to work with W3C and Jesse to get the [`solid/activitypub-interop`](https://github.com/solid/activitypub-interop) repository created.
```

### Wallet API
* (continuation if MdJ is present; MdJ mentioned work with Dutch universities)
* MdJ not present. Will delay until next week.

### ShapeRepo

> * SR: On Jesse's suggestion, I'd like to propose a Special Topic Meeting about a shared repository for shapes and shapetrees.
> * HZ: It's part of Solid mission to have such repositories.
> * SR: First to have a discussion and see where to go from there. I'll write a post on the Solid forum with requirements that I see.
>   I think we need to have at least Jesse and Jackson Morgan.
> * Rahul: Michiel de Jong has collated work on this, you might want to include it as well. https://pdsinterop.org/conventions/
> * Jesse: You should look at the [solid/roadmap repo](https://github.com/solid/roadmap)
> 
> ACTION: SR to prepare a short document with requirements and organize the STM meeting, with a poll to ensure people can be present.

Reviewer 3: ```suggestion
* SR: On Jesse's suggestion, I'd like to propose a Special Topic Meeting about a shared repository for shapes and shapetrees.
* HZ: It's part of the Solid mission to have such repositories.
* SR: First to have a discussion and see where to go from there. I'll write a post on the Solid forum with requirements that I see.
  I think we need to have at least Jesse and Jackson Morgan.
* Rahul: Michiel de Jong has collated work on this; you might want to include it as well. https://pdsinterop.org/conventions/
* Jesse: You should look at the [`solid/roadmap` repo](https://github.com/solid/roadmap)

ACTION: SR to prepare a short document with requirements and organize the STM meeting, with a poll to ensure people can be present.
```

> * RG: Are you still going forward with that grant proposal?
> * Jesse: There was a bounty proposal system, reviewers didn't understand how shapes work in Solid. I spoke with NLnet at FOSDEM, they are keen to support Solid and we could rework the grant and get it fast forwarded. We can discuss it as part of the STM.
> * eP: Will have a call with NLnet soon about a grant I asked before Oct 1st and it moved to phase 2, I will mention this to them.

Reviewer 3: ```suggestion
* RG: Are you still going forward with that grant proposal?
* Jesse: There was a bounty proposal system. Reviewers didn't understand how shapes work in Solid. I spoke with NLnet at FOSDEM; they are keen to support Solid and we could rework the grant and get it fast forwarded. We can discuss it as part of the STM.
* eP: I will have a call with NLnet soon about a grant. I asked before Oct 1st and it moved to phase 2; I will mention this to them.
```

### Insertion with Blank node with N3 Patch

https://github.com/CommunitySolidServer/CommunitySolidServer/issues/1968

https://github.com/solid/specification/issues/704

> * RG: Context: Migrating from NSS to CSS. NSS supports SPARQL update and N3 Patch, but CSS only supports N3 Patch. I don't know the context of the discussion, but want to know what I can do to make SolidOS work as it may be broken with only N3 Patch
> * Jesse: This seems like a discussion for the N3 CG, not the Solid CG.
> * eP: N3 is not so mature, can we really use it as a dependency?
> * Jesse: N3 is thinking about becoming a WG.
> * eP: N3-patch only exist within the Solid spec, so what will happen with the new WG ?
> * RG: We are not telling the WG to include this, but we are making them aware that there are some problems (?)
> * Jesse: If you have a clear solution in mind, no one is going to object that, and the WG will take this as a valuable input.
> * ...

Reviewer 3: ```suggestion
* RG: Context: Migrating from NSS to CSS. NSS supports SPARQL Update and N3 Patch, but CSS only supports N3 Patch. I don't know the context of the discussion, but want to know what I can do to make SolidOS work, as it may be broken with only N3 Patch.
* Jesse: This seems like a discussion for the N3 CG, not the Solid CG.
* eP: N3 is not so mature. Can we really use it as a dependency?
* Jesse: N3 is thinking about becoming a WG.
* eP: N3-patch only exists within the Solid spec, so what will happen with the new N3 WG?
* RG: We are not telling the WG to include this, but we are making them aware that there are some problems. (?)
* Jesse: If you have a clear solution in mind, no one is going to object to that, and the N3 WG will take this as a valuable input.
* ...
```

> * eP: If someone has a clear idea about why blank nodes should be allowed with N3 patch, they should create a PR so we can discuss on concrete terms with the community. And maybe warn the WG about this change.
> * Jesse: My proposal is to delete the "MUST NOT"
> * HZ: Minimal change could change *MUST NOT* to *SHOULD NOT*
> 
> ACTION: Jesse will create a PR removing that line, he will link to this discussion.

Reviewer 3: ```suggestion
* eP: If someone has a clear idea about why blank nodes should be allowed in N3 Patch, they should create a PR so we can discuss on concrete terms with the community. And maybe warn the N3 WG about this change.
* Jesse: My proposal is to delete the "MUST NOT".
* HZ: Minimal change could change *MUST NOT* to *SHOULD NOT*.

ACTION: Jesse will create a PR removing that line. He will link to this discussion.
```

### Misc

> * SR: I want to post discussion about profile management. It seems that Solid didn't think about it. The profiles are public, there is also a need for private profiles. If you share the full profile it can be problematic. The profile is the most sensitive data on the pod.
> * JW: There is a notion of extended profiles that are used by implementaiton. In public profile there are minimal information, there is a pointer to extended profile which are access protected.
> * SR: It is not only about public/private but about fine grained access. For now it is a single resource.
> * HZ: I think we will need to discuss views and contexts. 

Reviewer 3: ```suggestion
* SR: I want to post discussion about profile management. It seems that Solid didn't think about it. All profiles are now public. There is also a need for private profiles. If you share the full profile, it can be problematic. The profile is the most sensitive data on the pod.
* JW: There is a notion of extended profiles that are used by implementation. In public profiles, there is minimal information; there is a pointer to extended profiles, which are access protected.
* SR: It is not only about public/private, but also about fine grained access. For now, it is a single resource.
* HZ: I think we will need to discuss views and contexts. 
```
